webapp/sources/rudder/rudder-web/src/main/elm/healthcheck/sources/View.elm

```diff
@@ -0,0 +1,146 @@
+module View exposing (..)
+
+import DataTypes exposing (Check, Model, Msg(..), SeverityLevel(..))
+import Html exposing (Html, button, div, i, span, text)
+import Html.Attributes exposing (class)
+import Html.Events exposing (onClick)
+import List exposing (any, map, sortWith)
+import List.Extra exposing (minimumWith)
+import String
+
+compareSeverityLevel: SeverityLevel -> SeverityLevel -> Order
+compareSeverityLevel a b =
+  case (a, b) of
+      (Critical, Warning)        -> LT
+      (Critical, CheckPassed)    -> LT
+      (Warning, Critical)        -> GT
+      (Warning, CheckPassed)     -> LT
+      (CheckPassed, Warning)     -> GT
+      (CheckPassed, Critical)    -> GT
+      (Warning, Warning)         -> EQ
+      (CheckPassed, CheckPassed) -> EQ
+      (Critical, Critical)       -> EQ
+
+compareCheck: Check -> Check -> Order
+compareCheck a b =
+  let
+    orderLevel = compareSeverityLevel a.level b.level
+  in
+  case orderLevel of
+    EQ ->
+      if(a.msg > b.msg) then
+        GT
+      else
+        LT
+    _ -> orderLevel
+
+containsSeverityLevel: SeverityLevel -> List Check -> Bool
+containsSeverityLevel level checks =
+  any (\c -> c.level == level) checks
+
+chooseHigherSecurityLevel: List Check -> SeverityLevel
+chooseHigherSecurityLevel checks =
+  let
+    higherPriority = minimumWith (\c1 c2 -> compareSeverityLevel c1.level c2.level ) checks
```

Comment:
Seems to work with `minimumWith` but I probably missed something here, `maximumWith` return the first element which is a `CheckPassed`

Reply:
depends or your ordering yes ;) you changed the semantics between my comment (where critical was GT and is now LT) but It's ok to use minimumWith !!